lightning/src/ln/peers/handshake/mod.rs

```diff
@@ -392,3 +392,374 @@ impl PeerHandshake {
+
+		assert_matches!(test_ctx.inbound_handshake.process_act(&act1).unwrap(), (Some(Act::Two(_)), None));
+		assert_matches!(test_ctx.inbound_handshake.state, Some(HandshakeState::AwaitingActThree(_)));
+		assert!(test_ctx.inbound_handshake.get_remote_pubkey().is_none());
+	}
+
+	// Inbound::AwaitingActOne -> Error (bad version byte)
+	#[test]
+	fn peer_handshake_new_inbound_awaiting_act_one_bad_version() {
+		let mut test_ctx = TestCtx::new();
+		let mut act1 = do_process_act_or_panic!(test_ctx.outbound_handshake, &[]);
+		// Set bad version byte
+		act1[0] = 1;
+
+		assert_eq!(test_ctx.inbound_handshake.process_act(&act1).err(), Some(String::from("unexpected version")));
+	}
+
+	// Inbound::AwaitingActOne -> Error (invalid hmac)
+	#[test]
+	fn peer_handshake_new_inbound_awaiting_act_invalid_hmac() {
+		let mut test_ctx = TestCtx::new();
+		let mut act1 = do_process_act_or_panic!(test_ctx.outbound_handshake, &[]);
+		// corrupt the ciphertext
+		act1[34] = 0;
+
+		assert_eq!(test_ctx.inbound_handshake.process_act(&act1).err(), Some(String::from("invalid hmac")));
+	}
+
+	// Inbound::AwaitingActOne -> Error (invalid remote ephemeral key)
+	#[test]
+	fn peer_handshake_new_inbound_awaiting_act_invalid_remote_ephemeral_key() {
+		let mut test_ctx = TestCtx::new();
+		let mut act1 = do_process_act_or_panic!(test_ctx.outbound_handshake, &[]);
+		// corrupt the ephemeral public key
+		act1[1] = 0;
+
+		assert_eq!(test_ctx.inbound_handshake.process_act(&act1).err(), Some(String::from("invalid remote ephemeral public key")));
+	}
+
+	// Inbound::AwaitingActOne -> AwaitingActThree
+	#[test]
+	fn peer_handshake_new_inbound_awaiting_act_one_to_awaiting_act_three() {
+		let mut test_ctx = TestCtx::new();
+		let act1 = do_process_act_or_panic!(test_ctx.outbound_handshake, &[]);
+
+		assert_matches!(test_ctx.inbound_handshake.process_act(&act1).unwrap(), (Some(Act::Two(_)), None));
+		assert_matches!(test_ctx.inbound_handshake.state, Some(HandshakeState::AwaitingActThree(_)));
+		assert_eq!(test_ctx.inbound_handshake.get_remote_pubkey(), None);
+	}
+
+	// Outbound::AwaitingActTwo -> Complete (valid conduit)
+	#[test]
+	fn peer_handshake_outbound_awaiting_act_two_process() {
+		let mut test_ctx = TestCtx::new();
+		let act1 = do_process_act_or_panic!(test_ctx.outbound_handshake, &[]);
+		let act2 = do_process_act_or_panic!(test_ctx.inbound_handshake, &act1);
+
+		assert_matches!(test_ctx.outbound_handshake.process_act(&act2).unwrap(), (Some(Act::Three(_)), Some(_)));
+		assert_matches!(test_ctx.outbound_handshake.state, Some(HandshakeState::Complete));
+		assert_eq!(test_ctx.outbound_handshake.get_remote_pubkey(), Some(test_ctx.inbound_public_key));
+	}
+
+
+	// Outbound::AwaitingActTwo -> Complete (with extra data)
+	// Ensures that any remaining data in the read buffer is transferred to the conduit once
+	// the handshake is complete
+	// TODO: Is this valid? Don't we expect peers to need ActThree before sending additional data?
```

Comment:
@arik-so

Reply:
As discussed in Slack. It is probably best to view this first PR in the "files changed" view instead of commit-by-commit. Otherwise, you will see all of the bugs from 494 before they were fixed in subsequent commits.